lib/server/push.api.js

```diff
@@ -321,9 +321,49 @@ function querySend(query, raixNotification) {
 
   let firebasePromises = [];
 
-  Push.appTokens.find(query).forEach(function (appToken) {
+  let userVetActiveMap = new Map();
+  let vetMap = new Map();
+
+  let userIdsSet = new Set();
+  let vetIdentifiersSet = new Set();
+
+  let appTokens = Push.appTokens.find(query).fetch();
+
+  appTokens.forEach(function (appToken) {
+    if (appToken.userId) {
+      userIdsSet.add(appToken.userId);
+    }
+
+    vetIdentifiersSet.add(appToken.appName);
```

Comment:
@vec12 `appTokens` are already filtered by `appName`. Check `query` param in 
```
let appTokens = Push.appTokens.find(query).fetch();
```

So, you don't need `vetIdentifiersSet` as you already have `appIdentifier` in `appToken` document.